src/bundle-manifest.ts

```diff
@@ -150,6 +148,51 @@ export function generateBundles(depsIndex: TransitiveDependenciesMap):
 }
 
 /**
+ * Chains multiple bundle strategy functions together so the output of one
+ * becomes the input of the next and so-on.
+ */
+export function composeStrategies(strategies: BundleStrategy[]) {
+  return strategies.reduce((s1, s2) => (b) => s2(s1(b)));
+}
+
+/**
+ * Generates a strategy function which finds all non-entrypoint bundles which
+ * are dependencies of the given entrypoint and merges them into that
+ * entrypoint's bundle.
+ */
+export function generateEagerMergeStrategy(entrypoint: UrlString):
+    BundleStrategy {
+  return generateMatchMergeStrategy(
+      entrypoint,
+      b => b.entrypoints.has(entrypoint) && !getBundleEntrypoint(b));
+}
+
+/**
+ * Generates a strategy function which finds all bundles matching the predicate
+ * function and merges them into the bundle containing the target file.
+ */
+export function generateMatchMergeStrategy(
+    target: UrlString,
+    predicate: (matchBundle: Bundle, targetBundle: Bundle) =>
```

Comment:
Turns out that none of the uses of the `generateMatchMergeStrategy` wound up needing access to the `targetBundle` argument in the predicate (i.e. the bundle which contains the file referenced by the `target` argument of strategy generator.  I had thought they might, but nope.  So that's gone and now the predicate's remaining argument is unambiguous.